lib/server.go

```diff
@@ -1504,27 +1504,37 @@ func (srv *Server) _handleSnapshot(pp *Peer, msg *MsgDeSoSnapshotData) {
 	// already synced all the state corresponding to the sub-blockchain ending at the snapshot
 	// height, we will now mark all these blocks as processed. To do so, we will iterate through
 	// the blockNodes in the header chain and set them in the blockchain data structures.
-	err = srv.blockchain.db.Update(func(txn *badger.Txn) error {
-		for ii := uint64(1); ii <= srv.HyperSyncProgress.SnapshotMetadata.SnapshotBlockHeight; ii++ {
-			currentNode := srv.blockchain.bestHeaderChain[ii]
-			// Do not set the StatusBlockStored flag, because we still need to download the past blocks.
-			currentNode.Status |= StatusBlockProcessed
-			currentNode.Status |= StatusBlockValidated
-			srv.blockchain.addNewBlockNodeToBlockIndex(currentNode)
-			srv.blockchain.bestChainMap[*currentNode.Hash] = currentNode
-			srv.blockchain.bestChain = append(srv.blockchain.bestChain, currentNode)
-			err = PutHeightHashToNodeInfoWithTxn(txn, srv.snapshot, currentNode, false /*bitcoinNodes*/, srv.eventManager)
-			if err != nil {
-				return err
-			}
+	//
+	// We split the db update into batches of 10,000 block nodes to avoid a single transaction
+	// being too large and possibly causing an error in badger.
+	var blockNodeBatch []*BlockNode
+	for ii := uint64(1); ii <= srv.HyperSyncProgress.SnapshotMetadata.SnapshotBlockHeight; ii++ {
+		currentNode := srv.blockchain.bestHeaderChain[ii]
+		// Do not set the StatusBlockStored flag, because we still need to download the past blocks.
+		currentNode.Status |= StatusBlockProcessed
+		currentNode.Status |= StatusBlockValidated
```

Comment:
@tholonious - do we also need to set `currentNode.Status |= StatusBlockCommitted` if it's a PoS block? I think this is safe to do since we only store committed state in hypersync snapshots.

Reply:
Yes. Please set currentNode.Status |= StatusBlockCommitted here